addons/account/models/account_move.py

```diff
@@ -1494,6 +1494,13 @@ def _move_autocomplete_invoice_lines_create(self, vals_list):
             ctx_vals = {'default_type': vals.get('type') or self._context.get('default_type')}
             if vals.get('journal_id'):
                 ctx_vals['default_journal_id'] = vals['journal_id']
+                # reorder the companies in the context so that the company of the journal
+                # (which will be the company of the move) is the main one, ensuring all
+                # property fields are read with the correct company
+                journal_company = self.env['account.journal'].browse(vals['journal_id']).company_id
+                allowed_companies = self._context.get('allowed_company_ids', journal_company.ids)
+                reordered_companies = sorted(allowed_companies, key=lambda cid: cid != journal_company.id)
+                ctx_vals['allowed_company_ids'] = reordered_companies
```

Comment:
I suggest another way: use `with_company` everytime we access a property.

Reply:
I'll mae sure to do that i n 13.1 and master at the forward-port, but for in 13.0 `with_company` does'nt exist so I'm stuck with context manipulation.

Reply:
Also, I think it would be better to use `with_company` early-on and switch the context as soon as it makes sense in a flow rather than doing it at field access, which is a lot less readable and far more error-prone.